workflow/controller/operator.go

```diff
@@ -1906,8 +1906,15 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 			return retryParentNode, nil
 		}
 		retryParentNode = processedRetryParentNode
+		lastChildNode := getChildNodeIndex(retryParentNode, woc.wf.Status.Nodes, -1)
 		// The retry node might have completed by now.
 		if retryParentNode.Fulfilled() {
+			// If retry node has completed, set the output of the last child node to its output.
+			// Runtime parameters (e.g., `status`, `resourceDuration`) in the output will be used to emit metrics.
+			if lastChildNode != nil {
+				retryParentNode.Outputs = lastChildNode.Outputs.DeepCopy()
+				woc.wf.Status.Nodes[node.ID] = *retryParentNode
```

Comment:
does this need locking with mutex to prevent race?

Reply:
I think this probably doesn't require locking because the `woc.wf.Status.Nodes[node.ID] = *retryParentNode` only happens once the `retryParentNode.Fulfilled()` which indicates that either all retries are failed or one retry succeeds, and for both cases, we only set `woc.wf.Status.Nodes[node.ID]` once to the `lastChildNode`.

Also, the code is actually moved from https://github.com/argoproj/argo-workflows/pull/10489/files#diff-f321d4af83fcf8311dc80c0d50c59ac4c240f761206e7bb652709870eb9feb43L1925-L1928 to the current place because the `Outputs` are used for emitting metrics. Was it already a race condition before this PR?